main.go

```diff
@@ -89,12 +90,50 @@ func main() {
 		Namespace:              watchNamespace,
 	})
 	if err != nil {
-		setupLog.Error(err, "unable to start manager")
+		setupLog.Error(err, "unable to create manager")
 		os.Exit(1)
 	}
 
+	ctx := ctrl.SetupSignalHandler()
+
+	// Fetch ClientConfigs and create the clientCache
 	clientCache := clientcache.New(mgr.GetClient(), scheme)
 
+	cConfigs := k8ssandraiov1alpha1.ClientConfigList{}
+	err = mgr.GetClient().List(ctx, &cConfigs)
+	if err != nil {
+		setupLog.Error(err, "unable to fetch cluster connections")
+		os.Exit(1)
+	}
+
+	for _, cCfg := range cConfigs.Items {
+		// Create clients and add them to the client cache
+		cfg, err := clientCache.GetRestConfig(&cCfg)
+		if err != nil {
+			setupLog.Error(err, "unable to setup cluster connections")
+			os.Exit(1)
+		}
+
+		clientCache.CreateClient(cCfg.GetContextName(), cfg)
+
+		// Add cluster to the manager
+		c, err := cluster.New(cfg, func(o *cluster.Options) {
+			o.Scheme = scheme
+		})
+		if err != nil {
+			setupLog.Error(err, "unable to create manager cluster connection")
+			os.Exit(1)
+		}
+
+		err = mgr.Add(c)
+		if err != nil {
+			setupLog.Error(err, "unable to add cluster to manager")
+			os.Exit(1)
+		}
+	}
+
+	// Create the reconciler and start it
+
 	if err = (&controllers.K8ssandraClusterReconciler{
```

Comment:
With these changes in the client cache I think we can get rid of the `SetupMultiClusterWithManager` function and move the watches setup into `SetupWithManager`.